bin/accessibility-automation/helper.js

```diff
@@ -0,0 +1,215 @@
+const request = require('request');
+const os = require('os');
+const glob = require('glob');
+const helper = require('../helpers/helper');
+
+exports.checkAccessibilityPlatform = (user_config) => {
+  let accessibility = false;
+  user_config.browsers.forEach(browser => {
+    if (browser.accessibility) {
+      accessibility = true;
+      return true;
+    }
+  })
+  return accessibility;
+}
+
+exports.setAccessibilityCypressCapabilities = async (user_config, accessibilityResponse) => {
+  if (user_config.run_settings.accessibilityOptions) {
+
+  } else {
+    user_config.run_settings.accessibilityOptions = {}
+  }
+  user_config.run_settings.accessibilityOptions.authToken = accessibilityResponse.data.accessibilityToken;
+  user_config.run_settings.accessibilityOptions.auth = accessibilityResponse.data.accessibilityToken;
+  user_config.run_settings.accessibilityOptions.scannerVersion = accessibilityResponse.data.scannerVersion;
+  user_config.run_settings.system_env_vars.push(`ACCESSIBILITY_AUTH=${accessibilityResponse.data.accessibilityToken}`)
+  user_config.run_settings.system_env_vars.push(`ACCESSIBILITY_SCANNERVERSION=${accessibilityResponse.data.scannerVersion}`)
+}
+
+exports.createAccessibilityTestRun = async (user_config, framework) => {
+
+  const userName = user_config["auth"]["username"];
+  const accessKey = user_config["auth"]["access_key"];
+
+  try {
+    let settings = user_config.run_settings.accessibilityOptions;
+
+    const {
+      buildName,
+      projectName,
+      buildDescription
+    } = getBuildDetails(user_config);
+
+    const data = {
+      'projectName': projectName,
+      'buildName': buildName,
+      'startTime': (new Date()).toISOString(),
+      'description': buildDescription,
+      'source': {
+        frameworkName: "Cypress",
+        frameworkVersion: helper.getPackageVersion('cypress', user_config),
+        sdkVersion: helper.getAgentVersion()
+      },
+      'settings': settings,
+      'versionControl': await helper.getGitMetaData(),
+      'ciInfo':  helper.getCiInfo(),
+      'hostInfo': {
+        hostname: os.hostname(),
+        platform: os.platform(),
+        type: os.type(),
+        version: os.version(),
+        arch: os.arch()
+      },
+      'browserstackAutomation': process.env.BROWSERSTACK_AUTOMATION
+    };
+
+    const config = {
+      auth: {
+        user: userName,
+        pass: accessKey
+      },
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+
+    const response = await nodeRequest(
+      'POST', 'test_runs', data, config
+    );
+    logger.info("response in createAccessibilityTestRun", response);
+    process.env.BROWSERSTACK_TEST_ACCESSIBILITY = 'true';
+    process.env.BS_A11Y_JWT = response.data.data.accessibilityToken;
+    process.env.BS_A11Y_TEST_RUN_ID = response.data.data.id;
+  
+    this.setAccessibilityCypressCapabilities(user_config, response.data);
+    setAccessibilityEventListeners();
```

Comment:
Lets move this out, we should ideally set those value only is status Code is 200,

Reply:
Made the change, will set it true if I get the authtoken